crypto/keyring/keyring.go

```diff
@@ -203,8 +209,8 @@ type keystore struct {
 func newKeystore(kr keyring.Keyring, opts ...Option) keystore {
 	// Default options for keybase
 	options := Options{
-		SupportedAlgos:       SigningAlgoList{hd.Secp256k1},
-		SupportedAlgosLedger: SigningAlgoList{hd.Secp256k1},
+		SupportedAlgos:       DefaultSupportedAlgorithms,
+		SupportedAlgosLedger: DefaultSupportedLedgerAlgorithms,
```

Comment:
I don't think this is a right way to do it (especially by looking at the test). I'm against using global variables for changing the defaults. To be more clear:
* Package X shouldn't modify package Y globals  
* ideally global packages should be only constants (unless they are private, but even then they should be treated as constant if possible).

The original behavior is correct: use function (functional arguments) to modify the options.

Reply:
I agree, I'm not a fan of globals either and I've just tested a different way of defining keyring creation options without relying on them.

Will update the PR later today with such implementation.